src/main/java/pl/bykowsi/kurs/tydzien1pd/service/impl/ProShopService.java

```diff
@@ -0,0 +1,51 @@
+package pl.bykowsi.kurs.tydzien1pd.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import pl.bykowsi.kurs.tydzien1pd.configuration.LanguageSettings;
+import pl.bykowsi.kurs.tydzien1pd.model.PriceCalculationsData;
+import pl.bykowsi.kurs.tydzien1pd.model.Product;
+import pl.bykowsi.kurs.tydzien1pd.screeninfo.MessagesPrinter;
+import pl.bykowsi.kurs.tydzien1pd.service.Basket;
+import pl.bykowsi.kurs.tydzien1pd.service.ShopService;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Service
+@Profile("pro")
+public class ProShopService extends PlusShopService implements ShopService {
+
+    protected final BigDecimal discount;
+
+    @Autowired
+    public ProShopService(Basket basket, LanguageSettings languageSettings, PriceCalculationsData priceCalculationsData, @Value("${price.VAT}") Integer VAT, @Value("${price.discount}") Integer discount) {
+        super(basket, languageSettings, priceCalculationsData, VAT);
+        this.discount = BigDecimal.valueOf(discount);
+    }
+
+    @Override
+    public void calculateFinalPrice() {
+        List<Product> generatedBasket = CreateBasket(basket);
```

Comment:
to jest raczej lista produktów więc może nazwać to `products` jeśli chciałbyś mieć koszyk to należałoby utworzyć taką klasę która opakowywała by listę produktów